base/strings/basic.jl

```diff
@@ -525,7 +525,7 @@ function map(f, s::AbstractString)
 end
 
 function filter(f, s::AbstractString)
-    out = IOBuffer(StringVector(sizeof(s)), read=true, write=true)
+    out = IOBuffer(sizehint=sizeof(s), read=true, write=true)
```

Comment:
We no longer have to pass `read=true, write=true`, since this is the default when no `data` buffer is passed to the constructor.  (Similarly elsewhere.)